src/DeepCopy/Reflection/ReflectionHelper.php

```diff
@@ -21,16 +21,19 @@ public static function getProperties(\ReflectionClass $ref)
         $propsArr = array();
 
         foreach ($props as $prop) {
-            $f = $prop->getName();
-            $propsArr[$f] = $prop;
+            $propertyName = $prop->getName();
+            $propsArr[$propertyName] = $prop;
         }
 
         if ($parentClass = $ref->getParentClass()) {
             $parentPropsArr = self::getProperties($parentClass);
-            if (count($parentPropsArr) > 0) {
-                $propsArr = array_merge($parentPropsArr, $propsArr);
+            foreach ($propsArr as $key => $property) {
+                $parentPropsArr[$key] = $property;
             }
```

Comment:
:+1: 

I would suggest to put the test in the scope they are addressing, so into `tests/DeepCopyTest/Reflection/ReflectionHelperTest.php`. @mnapoli what do you think?